src/App.tsx

```diff
@@ -1,14 +1,33 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TodoFilter, getFilteredTodos } from './utils/TodoFilter';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(Filter.All);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState(0);
```

Comment:
This is number not a `Todo`
```suggestion
  const [selectedTodoId, setSelectedTodoId] = useState(0);
```